### PR TITLE
Fixes re-logging issues and Fixes playerUI appearing when ghosting

### PR DESCRIPTION
**Pull request by Bod9001**
### Changelog:

CL: [Fix] Fixes playerUI appearing when ghosting
CL: [Fix] Fixes re-logging issues where stuck on rejoining screen

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/Camera/CameraEffectControlScript.cs [hunk 1]
```diff
@@ -73,6 +85,18 @@ public void AddDrunkTime(float time)
 			}
 		}
 
+		public void BlindnessValue(bool IsBlind)
+		{
+			if (IsBlind)
+			{
+				Camera.main.GetComponent<CameraEffects.CameraEffectControlScript>().LightingSystem.fovDistance = 0.65f;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Camera/CameraEffectControlScript.cs]: make the numbers either a constant or serialized field.

File: UnityProject/Assets/Scripts/Core/Camera/CameraEffectControlScript.cs [hunk 2]
```diff
@@ -73,6 +85,18 @@ public void AddDrunkTime(float time)
 			}
 		}
 
+		public void BlindnessValue(bool IsBlind)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Camera/CameraEffectControlScript.cs]: add the summary. Also rename the parameter to ``isBlind`` with lower case ``i``.

File: UnityProject/Assets/Scripts/Core/Camera/CameraEffectControlScript.cs [hunk 3]
```diff
@@ -73,6 +85,18 @@ public void AddDrunkTime(float time)
...
+		{
+			if (IsBlind)
+			{
+				Camera.main.GetComponent<CameraEffects.CameraEffectControlScript>().LightingSystem.fovDistance = 0.65f;
+			}
+			else
+			{
+				Camera.main.GetComponent<CameraEffects.CameraEffectControlScript>().LightingSystem.fovDistance = 15;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Camera/CameraEffectControlScript.cs]: make the numbers either a constant or serialized field.

File: UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs [hunk 4]
```diff
@@ -160,10 +160,23 @@ public static GameObject RespawnPlayerAt(Mind mind, Occupation requestedOccupati
 	}
 
 
+	public static GameObject ClonePlayerAt(Mind mind, Occupation requestedOccupation, CharacterSheet character, Vector3? worldPos)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs:164]: add a summary for this public method.

File: UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs [hunk 5]
```diff
@@ -257,13 +270,27 @@ private static void ApplyNewSpawnRoleToBody( GameObject body, Occupation request
...
 
-		body.GetComponent<DynamicItemStorage>()?.SetUpOccupation(requestedOccupation);
+
 		//determine where to spawn them
 		physics.AppearAtWorldPositionServer(GetSpawnPointForOccupation(requestedOccupation));
 
 		switch (spawnType)
 		{
 			case SpawnType.NewSpawn:
+				body.GetComponent<DynamicItemStorage>()?.SetUpOccupation(requestedOccupation);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs]:
```suggestion
				body.GetComponent<DynamicItemStorage>().OrNull()?.SetUpOccupation(requestedOccupation);
```

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs]: this switch looks rather busy, maybe move the body of the cases to private methods?

File: UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs [hunk 6]
```diff
@@ -257,13 +270,27 @@ private static void ApplyNewSpawnRoleToBody( GameObject body, Occupation request
...
+						body,
+						requestedOccupation.DisplayName,
+						requestedOccupation.SpawnSound.AssetAddress,
+						requestedOccupation.TextColor,
+						requestedOccupation.BackgroundColor,
+						requestedOccupation.PlaySound);
+				}
+				break;
+			case SpawnType.ReSpawn:
+				body.GetComponent<DynamicItemStorage>()?.SetUpOccupation(requestedOccupation);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs]:
```suggestion
				body.GetComponent<DynamicItemStorage>().OrNull()?.SetUpOccupation(requestedOccupation);
```

**Review by corp-0 — CHANGES_REQUESTED** [review]

File: UnityProject/Assets/Scripts/Core/Camera/CameraEffectControlScript.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Camera/CameraEffectControlScript.cs]: looks like this accidentally got included

**Comment by corp-0** [comment]
Make sure to use past tense in the changelog. Also elaborate on the second one, what issues did it fix?

**Pull request merged**